src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/reader/MetricsEmitter.java

```diff
@@ -847,4 +848,141 @@ public static void emitNodeMetrics(
           "Total time taken for writing {} metrics metricsdb: {}", tableName, mFinalT - mCurrT);
     }
   }
+
+    public static void emitFaultDetectionMetrics(DSLContext create, MetricsDB db, FaultDetectionMetricsSnapshot faultDetectionSnapshot) {
+
+      long mCurrT = System.currentTimeMillis();
+      Dimensions dimensions = new Dimensions();
```

Comment:
Replace this with `List<String> dimensionNames = snap.getDimensionNames();` ?

Reply:
It is a map of dimension name and its value. It will be constructed once we have the table of records